lib/core/logger/logger.provider.js

```diff
@@ -1,21 +1,14 @@
 'use strict';
 
 const Logger = require('./logger').Logger;
-const LOGGER_SYMBOL_KEY = require('./symbol.constant').loggerKey;
+const BaseProvider = require('../../public-api/public-api').BaseProvider;
 
-function provider(level = null) {
-    const loggerSymbol = Symbol.for(LOGGER_SYMBOL_KEY);
-
-    if (global[loggerSymbol]) {
-        // being here means that an instance of the logger already exist
-        return global[loggerSymbol];
-    } else {
-        return (global[Symbol.for(LOGGER_SYMBOL_KEY)] = new Logger(level));
+class LoggerProvider extends BaseProvider {
+    constructor() {
+        super('logger', Logger);
```

Comment:
why we don't simplify it so a line like in the baseprovider-test like:
```suggestion
const loggerProvider = new BaseProvider('logger', Logger);
```
So we won't need to call the provider with a `new` and we don't need to call any constructor.

Reply:
I am not sure I understand this suggestion here, could you elaborate?